actions/postgresql_show_create_table_action.go

```diff
@@ -0,0 +1,466 @@
+       (SELECT substring(pg_catalog.pg_get_expr(d.adbin, d.adrelid) for 128)
+        FROM pg_catalog.pg_attrdef d
+        WHERE d.adrelid = a.attrelid
+          AND d.adnum = a.attnum
+          AND a.atthasdef),
+       a.attnotnull,
+       (SELECT c.collname
+        FROM pg_catalog.pg_collation c,
+             pg_catalog.pg_type t
+        WHERE c.oid = a.attcollation
+          AND t.oid = a.atttypid
+          AND a.attcollation <> t.typcollation)                          AS attcollation,
+       ''::pg_catalog.char                                               AS attidentity,
+       a.attstorage,
+       CASE WHEN a.attstattarget = -1 THEN NULL ELSE a.attstattarget END AS attstattarget,
+       pg_catalog.col_description(a.attrelid, a.attnum)
+FROM pg_catalog.pg_attribute a
+WHERE a.attrelid = $1
+  AND a.attnum > 0
+  AND NOT a.attisdropped
+ORDER BY a.attnum;`, tableID)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	tw := tabwriter.NewWriter(w, 0, 0, 1, ' ', tabwriter.Debug)
+
+	fmt.Fprintln(tw, "Column\tType\tCollation\tNullable\tDefault\tStorage\tStats target\tDescription") //nolint:errcheck
+
+	for rows.Next() {
+		var ci columnInfo
+		err = rows.Scan(
+			&ci.Attname,
+			&ci.FormatType,
+			&ci.Substring,
+			&ci.Attnotnull,
+			&ci.Attcollation,
+			&ci.Attidentity,
+			&ci.Attstorage,
+			&ci.Attstattarget,
+			&ci.ColDescription,
+		)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		fmt.Fprintf(tw, "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n",
+			ci.Attname,
+			ci.FormatType,
+			pointer.GetString(ci.Attcollation),
+			formatNullable(ci.Attnotnull),
+			pointer.GetString(ci.Substring),
+			formatStorage(ci.Attstorage),
+			pointer.GetString(ci.Attstattarget),
+			pointer.GetString(ci.ColDescription),
+		) //nolint:errcheck
+	}
+	err = rows.Err()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	return tw.Flush()
+}
+
+func (a *postgresqlShowCreateTableAction) printIndexInfo(ctx context.Context, w io.Writer, db *sql.DB, tableID string) error {
+	rows, err := db.QueryContext(ctx, `SELECT /* pmm-agent */  c2.relname,
+       i.indisprimary,
+       i.indisunique,
+       i.indisclustered,
+       i.indisvalid,
+       i.indrelid::regclass,
+       pg_catalog.pg_get_indexdef(i.indexrelid, 0, false),
+       pg_catalog.pg_get_constraintdef(con.oid, true),
+       contype,
+       condeferrable,
+       condeferred,
+       i.indisreplident,
+       c2.reltablespace
+FROM pg_catalog.pg_class c,
+     pg_catalog.pg_class c2,
+     pg_catalog.pg_index i
+         LEFT JOIN pg_catalog.pg_constraint con
+                   ON (conrelid = i.indrelid AND conindid = i.indexrelid AND contype IN ('p', 'u', 'x'))
+WHERE c.oid = $1
+  AND c.oid = i.indrelid
+  AND i.indexrelid = c2.oid
+ORDER BY i.indisprimary DESC, i.indisunique DESC, c2.relname`, tableID)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	defer rows.Close()
+
+	var buf bytes.Buffer
+	bw := bufio.NewWriter(&buf)
```

Comment:
We don't need it

Reply:
Actually, we need it here to use `Flush` method.
We need `Flush` method to not print header if there are no rows.

Reply:
Well, then please add a comment where you call Flush, and a test. Otherwise, it looks like a simple mistake, not a feature.